pkg/api/customization/catalog/template.go

```diff
@@ -2,23 +2,34 @@ package catalog
 
 import (
 	"bytes"
+	"fmt"
 	"net/http"
+	"strconv"
 	"strings"
 	"time"
 
 	"github.com/rancher/norman/api/access"
 	"github.com/rancher/norman/httperror"
 	"github.com/rancher/norman/types"
 	helmlib "github.com/rancher/rancher/pkg/catalog/helm"
+	catUtil "github.com/rancher/rancher/pkg/catalog/utils"
+	hcommon "github.com/rancher/rancher/pkg/controllers/user/helm/common"
 	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
 	managementschema "github.com/rancher/types/apis/management.cattle.io/v3/schema"
 	client "github.com/rancher/types/client/management/v3"
 )
 
-func TemplateFormatter(apiContext *types.APIContext, resource *types.RawResource) {
+type TemplateWrapper struct {
+	CatalogLister                v3.CatalogLister
+	ClusterCatalogLister         v3.ClusterCatalogLister
+	ProjectCatalogLister         v3.ProjectCatalogLister
+	CatalogTemplateVersionLister v3.CatalogTemplateVersionLister
+}
+
+func (t TemplateWrapper) TemplateFormatter(apiContext *types.APIContext, resource *types.RawResource) {
 	var prjCatalogName, clusterCatalogName string
 	// version links
-	resource.Values["versionLinks"] = extractVersionLinks(apiContext, resource)
+	resource.Values["versionLinks"] = t.extractVersionLinks(apiContext, resource)
```

Comment:
Hmm, why the choice to use a receiver here and why not use a pointer receiver?

Reply:
`extractVersionLinks` needs access to `templateVersionForRancherVersion` which needs `CatalogTemplateVersionLister`. 

Non pointer as none of the other methods are and we are not modifying values on `TemplateWrapper`.